airflow/providers/amazon/aws/hooks/s3.py

```diff
@@ -57,8 +57,14 @@ def wrapper(*args, **kwargs) -> T:
 
         if "bucket_name" not in bound_args.arguments:
             self = args[0]
-            if self.conn_config and self.conn_config.schema:
-                bound_args.arguments["bucket_name"] = self.conn_config.schema
+            if self.aws_conn_id:
+                conn_args = self.get_connection(self.aws_conn_id).extra_dejson
```

Comment:
Get connection it is quite expensive call there is no reason call it every time when decorated method called.

IMHO better:
1. create new property/attribute in [AwsConnectionWrapper](https://github.com/apache/airflow/blob/8290ade26deba02ca6cf3d8254981b31cf89ee5b/airflow/providers/amazon/aws/utils/connection_wrapper.py#L76) which store this dictionary

2. Add method to `get_service_config(service_name: str)` to AwsConnectionWrapper which return config for specific config service
```python

def get_service_config(service_name: str) -> dict:
    return self.service_config.get(service_name, {})
```

3. Add property into [AwsGenericHook](https://github.com/apache/airflow/blob/8290ade26deba02ca6cf3d8254981b31cf89ee5b/airflow/providers/amazon/aws/hooks/base_aws.py#L379) which associated with selected `service_name`

```python

@property
def service_config() -> dict:
    service_name = client_type or resource_type
    return self.conn_config.get_service_config(service_name)
```

In this case we only call get_connection once because `AwsGenericHook.conn_config` is cached property.

Reply:
This also make integration with other services easier

Reply:
And you need only call in decorator something like this

```python

if "bucket_name" in self.service_config:
    ...

```

Reply:
That is a good point. I have added the property `service_config` to `AwsGenericHook`, and now the conditional logic within `provide_bucket_name` is a lot simpler, and it utilizes caching! Thank you for the suggestion.